manifest-build-tools/application/make_debian_packages.py

```diff
@@ -0,0 +1,232 @@
+
+The optional parameters:
+is-official-release: Whether this release is official. The default value is False
+parameter-file: The file with parameters. The file will be passed to downstream jobs.
+force: Use destination directory, even if it exists.
+sudo-credential: The environment variable name of sudo credentials.
+                 For example: SUDO_CRED=username:password
+jobs: Number of parallel jobs(build debian packages) to run.
+      The number is related to the compute architecture, multi-core processors..
+force:
+"""
+
+import argparse
+import os
+import sys
+import json
+
+try:
+    from reprove import ManifestActions
+    from update_rackhd_version import RackhdDebianControlUpdater
+    from version_generator import VersionGenerator
+    from DebianBuilder import DebianBuilder
+    import common
+except ImportError as import_err:
+    print import_err
+    sys.exit(1)
+
+def parse_args(args):
+    """
+    Parse script arguments.
+    :return: Parsed args for assignment
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--build-directory',
+                        required=True,
+                        help="Top level directory that stores all the cloned repositories.",
+                        action='store')
+
+    parser.add_argument('--manifest-file',
+                        required=True,
+                        help="The file path of manifest",
+                        action='store')
+
+    parser.add_argument('--parameter-file',
+                        help="The jenkins parameter file that will used for succeeding Jenkins job",
+                        action='store',
+                        default="downstream_parameters")
+
+    parser.add_argument('--git-credential',
+                        required=True,
+                        help="Git URL and credential for CI services: <URL>,<Credentials>",
+                        action='append',
+                        default=None)
+
+    parser.add_argument('--sudo-credential',
+                        help="username:password pair for sudo user",
+                        action='store',
+                        default=None)
+
+    parser.add_argument('--jobs',
+                        help="Number of build jobs to run in parallel",
+                        default=-1,
+                        type=int,
+                        action="store")
+
+    parser.add_argument('--is-official-release',
+                        default="false",
+                        help="Whether this release is official",
+                        action="store")
+
+    parser.add_argument('--force',
+                        help="Overwrite a directory even if it exists",
+                        action="store_true")
+
+    parsed_args = parser.parse_args(args)
+    parsed_args.is_official_release = common.str2bool(parsed_args.is_official_release)
+    return parsed_args
+
+def update_rackhd_control(top_level_dir, is_official_release):
```

Comment:
please add verbose comment for this function.
what it's used for ..
how it's implemented..
from where  RackHD gets sub-on-xxx's deb package version